cmake/SpectreSetupPythonPackage.cmake

```diff
@@ -33,6 +33,14 @@ configure_or_symlink_py_file(
 )
 # Also link the main entry point to bin/
 set(PYTHON_SCRIPT_LOCATION "spectre")
+set(JEMALLOC_PRELOAD "")
+if(BUILD_PYTHON_BINDINGS AND "${JEMALLOC_LIB_TYPE}" STREQUAL SHARED)
+  if(DEFINED ENV{LD_PRELOAD})
```

Comment:
I'd be OK with the old version that didn't respect LD_PRELOAD at all, but if you are going to respect it you need to use the value in the script's environment, not cmake's environmment.